File: C++/Hard/RuntimeError/minimum-difficulty-of-a-job-schedule_9.cpp
class Solution {
public:
    int minDifficulty(vector<int>& jobDifficulty, int d) {
        
        int n = jobDifficulty.size();
        if (n < d) return -1;
        vector<vector<int>> dp(n, vector<int>(d, 0));
        int maxv = jobDifficulty[0];
        for (int i = 0; i < d; i++) {
            maxv = max(maxv, jobDifficulty[i]);
            dp[i][i] = maxv;
        }
        for (int i = 1; i < n; i++) {
            for (int j = 0; j < min(i, d-1); j++) {
                int maxv = jobDifficulty[i];
                for (int k = i-1; k >= j; k--) {
                    maxv = max(maxv, jobDifficulty[k]);
                    dp[i][j] = max(dp[i][j], dp[k-1][j-1] + maxv);
                }
            }
        }
        return dp[n-1][d-1];
    }
};

File: Rust/Hard/WrongAnswer/minimum-difficulty-of-a-job-schedule_9.rs
impl Solution {
    pub fn min_difficulty(job_difficulty: Vec<i32>, d: i32) -> i32 {
        
        let d = d as usize;
        let mut dp = vec![vec![-1; job_difficulty.len()+1]; d+1];
        dp[0][0] = 0;
        for i in 1..=d {
            for j in i..=job_difficulty.len() {
                let mut max = 0;
                for k in (i-1)..j {
                    max = max.max(job_difficulty[k]);
                    if dp[i-1][k] != -1 {
                        dp[i][j] = dp[i][j].min(dp[i-1][k] + max).max(-1);
                    }
                }
            }
        }
        dp[d][job_difficulty.len()]
    }
}